src/com/owncloud/android/ui/preview/FileDownloadFragment.java

```diff
@@ -226,117 +202,124 @@ public void onClick(View v) {
-        TextView progressText = (TextView) getView().findViewById(R.id.progressText);
-        progressText.setText(R.string.downloader_download_in_progress_ticker);
-        progressText.setVisibility(View.VISIBLE);
-
-        // hides the error icon
-        getView().findViewById(R.id.errorText).setVisibility(View.GONE);
-        getView().findViewById(R.id.error_image).setVisibility(View.GONE);
+    private void setButtonsForTransferring(View rootView) {
+        if (rootView != null) {
+            rootView.findViewById(R.id.cancelBtn).setVisibility(View.VISIBLE);
+
+            // show the progress bar for the transfer
+            rootView.findViewById(R.id.progressBar).setVisibility(View.VISIBLE);
+            TextView progressText = (TextView) rootView.findViewById(R.id.progressText);
+            progressText.setText(R.string.downloader_download_in_progress_ticker);
+            progressText.setVisibility(View.VISIBLE);
+
+            // hides the error icon
+            rootView.findViewById(R.id.errorText).setVisibility(View.GONE);
+            rootView.findViewById(R.id.error_image).setVisibility(View.GONE);
+        }
     }
 
+
     /**
      * Enables or disables buttons for a file locally available
      */
-    private void setButtonsForDown() {
-        getView().findViewById(R.id.cancelBtn).setVisibility(View.GONE);
+    private void setButtonsForDown(View rootView) {
+        if (rootView != null) {
+            rootView.findViewById(R.id.cancelBtn).setVisibility(View.GONE);
 
-        // hides the progress bar
-        getView().findViewById(R.id.progressBar).setVisibility(View.GONE);
+            // hides the progress bar
+            rootView.findViewById(R.id.progressBar).setVisibility(View.GONE);
 
-        // updates the text message
-        TextView progressText = (TextView) getView().findViewById(R.id.progressText);
-        progressText.setText(R.string.common_loading);
-        progressText.setVisibility(View.VISIBLE);
+            // updates the text message
+            TextView progressText = (TextView) rootView.findViewById(R.id.progressText);
+            progressText.setText(R.string.common_loading);
+            progressText.setVisibility(View.VISIBLE);
 
-        // hides the error icon
-        getView().findViewById(R.id.errorText).setVisibility(View.GONE);
-        getView().findViewById(R.id.error_image).setVisibility(View.GONE);
+            // hides the error icon
+            rootView.findViewById(R.id.errorText).setVisibility(View.GONE);
+            rootView.findViewById(R.id.error_image).setVisibility(View.GONE);
+        }
     }
 
-
     /**
      * Enables or disables buttons for a file not locally available
      * <p/>
      * Currently, this is only used when a download was failed
      */
-    private void setButtonsForRemote() {
-        getView().findViewById(R.id.cancelBtn).setVisibility(View.GONE);
+    private void setButtonsForRemote(View rootView) {
```

Comment:
mostly the same code as in setButtonsForDown but with different visibility states to be set, this could probably be abstracted.